test/hotspot/jtreg/gc/shenandoah/mxbeans/TestChurnNotifications.java

```diff
@@ -159,7 +159,12 @@ public void handleNotification(Notification n, Object o) {
 
         System.gc();
 
-        Thread.sleep(1000);
+        // Wait until notifications start arriving, and then wait some more
+        // to catch the ones arriving late.
+        while (churnBytes.get() == 0) {
+            Thread.sleep(1000);
+        }
+        Thread.sleep(5000);
```

Comment:
I was wandering if it's necessary to handle the timeout by the code itself instead of delegating to jtreg? In the worst case, that's a really a long time about 960000ms.

Reply:
Yeah, I was wondering about the same when doing that chunk, but then argued to myself that timeout might be as well handled by jtreg. Mostly because in many cases JTREG_TIMEOUT_FACTOR is passed to control the machine-dependent behavior: slower/overloaded machines get larger timeout factors configured. Hardcoding the timeouts in the test would deprive us of this "feature".

Reply:
Thanks, that's make sense to me.